src/app/sitemap.ts

```diff
@@ -1,25 +1,42 @@
 import type { MetadataRoute } from "next";
 
 import { client } from "studio/lib/client";
-import { Slug } from "studio/lib/interfaces/global";
+import { DocumentWithSlug } from "studio/lib/interfaces/global";
+import { PageBuilder } from "studio/lib/interfaces/pages";
+import { DOCUMENTS_WITH_SLUG_QUERY } from "studio/lib/queries/document";
+import { LANDING_PAGE_QUERY } from "studio/lib/queries/navigation";
 import { token } from "studio/lib/token";
 
-interface SitemapDocument {
-  slug: Slug;
-  _updatedAt: string;
-}
-
 const clientWithToken = client.withConfig({ token });
 
 export const dynamic = "force-dynamic";
 export const fetchCache = "default-no-store";
 
 export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
-  const documents =
-    await clientWithToken.fetch<SitemapDocument[]>(`*[defined(slug)]`);
-
-  return documents.map((s) => ({
-    url: new URL(s.slug.current, process.env.NEXT_PUBLIC_URL).toString(),
+  const baseUrl =
+    process.env.NEXT_PUBLIC_URL !== undefined &&
+    URL.canParse(process.env.NEXT_PUBLIC_URL)
+      ? new URL(process.env.NEXT_PUBLIC_URL)
+      : undefined;
+  if (baseUrl === undefined) {
+    console.error("Failed to generate sitemap, missing baseUrl");
+    return [];
+  }
+  const slugDocuments = await clientWithToken.fetch<DocumentWithSlug[]>(
+    DOCUMENTS_WITH_SLUG_QUERY,
+  );
+  const sitemapEntries = slugDocuments.map((s) => ({
```

Comment:
I'm not really a fan of single letters just because it's easy to lose track of context when reviewing or jumping into someoneelses code.
```suggestion
  const sitemapEntries = slugDocuments.map((document | slugDocument) => ({
```